charts/cortex-agent/templates/daemonset.yaml

```diff
@@ -172,5 +172,7 @@ spec:
           secretName: {{ include "cortex-xdr.deploymentSecretName" . }}
       {{- end }}
 
+      {{- if .Values.dockerPullSecretName }}
```

Comment:
Where is this value being set? Did you maybe mean `{{- if .Values.dockerPullSecret.name }}`?

Reply:
> Where is this value being set? Did you maybe mean `{{- if .Values.dockerPullSecret.name }}`? if we want to check that the secret name is not empty we can do: `{{- if (include "cortex-xdr.deploymentSecretName" .) }}`

Currently, the conditional is
```
{{- if .Values.dockerPullSecret.create }}
```
which breaks the chart for those who want to provide their own secret via `.Values.dockerPullSecret.name.`

What about updating it to
```
{{- if or .Values.dockerPullSecret.create .Values.dockerPullSecret.name }}
```
?

Reply:
My suggestion should work for when both are not set, because `(include "cortex-xdr.deploymentSecretName" .)` will be an empty string (will get the value of `.Values.dockerPullSecret.name`) so we won't render `imagePullSecrets`. 
`cortex-xdr.deploymentSecretName` takes into account those 2 values when defined in `_helpers.tpl`.
Are we fine with that?

Reply:
I tried your suggestion and it does not work. With `.Values.dockerPullSecret.create` set to false and  `.Values.dockerPullSecret.name` unset, it still renders
```
      imagePullSecrets:
      - name:
```

Reply:
yea my bad it should be `{{- if (include "cortex-xdr.dockerPullSecretName" .) }}`, just tested it. can you confirm?

Reply:
Cool. Since the company policy doesn't allow me to merge PRs from outside of the org, i will make a PR myself with the same fix.
Thanks!

Reply:
@nickperry https://github.com/PaloAltoNetworks/cortex-helm/pull/29
this will be merged to master and then go into 1.6.2